### PR TITLE
EntityTitleService uses SearchUser instead of Subject

### PR DESCRIPTION
**Pull request by todvora**
EntityTitleService used Subject, which is quite complex and heavy-weight object. With the SearchUser and especially with its Permission interfaces we can test the code more easily and don't need that many mocks. 

Extracted interface for cleaner access to the service.

/nocl

## Description
Purely internal refacoring

## How Has This Been Tested?
Existing unit and integration tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by luk-kaminski** [comment]
Wouldn't it be easier to store Subject as a field in SearchUser, with some getter?
With this approach one could use the service, no matter if having Subject object or SearchUser object wrapping Subject object.
Now no one can user the service with Subject anymore.

**Comment by todvora** [comment]
@luk-kaminski The SearchUser is an abstraction that helps us to get rid of the Subject dependency. We don't want to use Subject in our implementations because it's too powerful and broad, needs to be mocked (fragile) and doesn't guide you into correct implementations and usages. 

But the service itself doesn't accept SearchUser, it accepts TitlePermissions, very narrow and specific interface. So you could always call the service with the original subject like this, using the Subject instance:

```java
service.getTitles((perm, id -> getSubject().isPermitted(perm + ":" + id), request);
```

The goal here is not to replace Subject with SearchUser but to narrow down the needed dependency to very basic and limited TitlePermissions interface which happens to be also implemented by the SearchUser. 

This limits the error surface, communicates better what's the real dependency, helps testing and makes code more cleaner and resilient.

File: graylog2-server/src/test/java/org/graylog2/rest/resources/system/contentpacks/CatalogResourceTest.java [hunk 1]
```diff
@@ -139,9 +140,9 @@ public void getTitles() {
         );
         doReturn(expectedResponse)
                 .when(entityTitleService)
-                .getTitles(any(), eq(request));
+                .getTitles(eq(request), any());
```
Comment (luk-kaminski) [inline graylog2-server/src/test/java/org/graylog2/rest/resources/system/contentpacks/CatalogResourceTest.java:143]: It is very minor, but I would replace `any()` with `eq(<test search user instance>)`

**Review by luk-kaminski — APPROVED** [review]
LGTM, thx!

**Pull request merged**